packages/nextjs/hooks/scaffold-eth/useDisplayUsdMode.ts

```diff
@@ -0,0 +1,21 @@
+import { useCallback, useEffect, useState } from "react";
+import { useGlobalState } from "~~/services/store/store";
+
+export const useDisplayUsdMode = ({ defaultUsdMode = false }: { defaultUsdMode?: boolean }) => {
+  const nativeCurrencyPrice = useGlobalState(state => state.nativeCurrency.price);
+  const isPriceFetched = nativeCurrencyPrice > 0;
+  const predefinedUsdMode = isPriceFetched ? Boolean(defaultUsdMode) : false;
+  const [displayUsdMode, setDisplayUsdMode] = useState(predefinedUsdMode);
+
+  useEffect(() => {
+    setDisplayUsdMode(predefinedUsdMode);
+  }, [predefinedUsdMode]);
```

Comment:
Also I changed this `useEffect` dependency, so it does not depend on price and not trigger on every price change, and hence work with `pollingInterval`. See https://github.com/scaffold-eth/scaffold-eth-2/pull/856#discussion_r1619316376

Reply:
Niccee !!! Just to be sure I tested by adding this to `useNaticeCurrencyPrice` hook : 

```ts
  useEffect(() => {
    // Increase the price by 1000 every 2 seconds
    const interval = setInterval(() => {
      setNativeCurrencyPrice(price => price + 1000);
    }, 2000);

    return () => clearInterval(interval);
  }, [nativeCurrencyPrice]);
```

And it doesn't trigger this hook